### ISSUE TITLE
user Story 3 - Create Project

### ISSUE THREAD

**Issue by abaibik** [UmaySoyluer/capstone-project#5]
## Value proposition
As a team leader I want to be able to create a new project for different customers
## Description

![Image](https://github.com/UmaySoyluer/capstone-project/assets/78874362/e3cab8f8-0d8b-40c2-a01f-d5a33aa58193)

## Acceptance criteria
- the create-button is displayed 
- clicking the create-button navigates to create project page where the user can describe a title , a description and the other informations for the new project
- create project page , there is displayed a form with 
input - name
input - description
input - start date
input - end date
input - project lead
input - select - department
submit button
cancel button

- after submitting the form , the new project will appended to the current projects and user will be navigated back to the Projects Overview Page
- Clicking cancel-button closes the form without submitting and navigates back
## Tasks
- create the component Button inside ui/Button.js
- create the component Form inside ui/Form.js
- Form has six inputs , a submit button and a cancel button 
- Form props:
formName
onSubmit

- render the Form component in pages/projects/new-project.js
- implement a handleSubmit function that creates a new project 
- updates the data on mongodb
- implement a handleClick function that navigates the page to Projects Overview Page
- add a router.push(“/”) inside handleSubmit function

**Comment by FelineHuhn** [comment]
Another great user story 😍 I give you some more suggestions. 

**Value Proposition:**
- since we don't have different roles for now, it would be more suitable when you write As a team member (like before, and the team lead is also a team member, so it's totally fine for now) 

**Description:**
- An arrow would be again helpful, to show that the user navigates to the create form when clicking the project card
- is the realistic headline of the create page "Create Form"?
- For a better user experience the required inputs could marked as required (e.g. with an `*`)

*Acceptance criteria:**
- for the first line, the create button could be more described, like that it has the same format as the project cards and is on the first place on the top and displays a plus icon
- on the create page the user cannot only "describe" but can fill input fields with information, or? Also "other informations for the new project" is not enough description for an acceptance criteria i think. You could combine similar inputs for a shorter user story, maybe I give you an example: 
- clicking the create button navigates to a page with the headline "Create Form" (or another?) 
  - a form with several input fields, in which the user can fill information about a new project, is displayed:
    - the required inputs are: name, description, start-date, end-date
    - the ...
- where is the new project located in the overview after submitting, and the end? At the top?
- Cancel button navigates back "to overview page" or something similar would would be a bit more clear

**Tasks:**
- You could remove the folder structure / path mentions, since it is not that important for a user story

**Comment by miriam-ertl** [comment]
Thank You for letting me look at your User Story, please consider and implement the suggestions of Feline. 

Due to the suggested AC of Feline
- [ ] Cancel button navigates back "to overview page" or something similar would would be a bit more clear
You could think about integrating a component like a `ConfirmMessage` or an alert message asking the team member e.g. "Do you really want to cancel the process?" or something like that. This could be another user story.

**Comment by FelineHuhn** [comment]
Good idea with the ConfirmMessage, but could be part of a future feature :) Well done!
